types/superagent/superagent-tests.ts

```diff
@@ -282,8 +282,7 @@ request
     .get("/blob")
     .responseType("blob")
     .end((err, res) => {
-        assert(res.xhr instanceof XMLHttpRequest);
-        assert(res.xhr.response instanceof Blob);
```

Comment:
Would this work?

````ts
if ("Blob" in globalThis) {
   assert(res.xhr.response instanceof (globalThis as any).Blob);
}
````

I *think* that would give you the top level browser Blob native constructor at runtime, without TS trying to actually type it.

Reply:
I'm not sure I understand the value of this. Since you've typed `globalThis` as `any`, you're just asserting that `res.xhr.respnose` is an instance of `any`, which we already know it is since I've typed `res.xhr` as `any`.

Reply:
I may have crossed wires; if so, please set me straight.

I've used other projects that have a type-testing framework, for example [this PR I submitted to Pino](https://github.com/pinojs/pino/pull/1195/files) which uses the `expectType` method from `tsd` to test that a given expression evaluates to a given type, using an explicit generic argument.  This happens at the type-checker level.  I don't think that's what's happening here.

It looks like you're using the Node `assert` module to actually make a request, then examine its response and fail if the `response` value isn't an instance of the global Blob contructor.  This appears to happen at runtime -- the test file is compiled to JS and run, presumably in browser.  That's why I was casting any-casting to reach `Blob`, I don't care what the typechecker thinks `xhr.response` is but I thought you were making a runtime check that it actually is a Blob.

If you're not making runtime checks, `assert` calls are not useful.  You could use a framework like `tsd` and write `expectType<Blob>(res.xhr.response)`, or simply write `const _res: Blob = res.xhr.response;`.  The typechecker would flag it as an incompatible assignment, which is not the case for the `assert` calls.

Reply:
The code in the `DefinitelyTyped` test suites isn't actually run; it's just compiled to check the TypeScript definitions all check out.

You can happily write:

```ts
assert(false);
```

And the tests will pass.

This is why I just removed the `assert()` line (which doesn't actually assert anything), and replaced it with a `console.log()`, which just lets us flex the fact that we're accessing a proprety on `res.xhr`.

To be honest I could probably even drop the `console.log()`; I just do it by habit, because a single line with a prop access and no assignment doesn't look like real code to me.